demos/smart_classroom_demo_gapi/main.cpp

```diff
@@ -0,0 +1,561 @@
+        const auto fr_model_path = FLAGS_m_reid;
+        const auto lm_model_path = FLAGS_m_lm;
+        const auto teacher_id = FLAGS_teacher_id;
+        const auto ids_list = FLAGS_fg;
+
+        if (!FLAGS_teacher_id.empty() && !FLAGS_top_id.empty()) {
+            slog::err << "Cannot run simultaneously teacher action and top-k students recognition."
+                      << slog::endl;
+            return 1;
+        }
+
+        const auto actions_type = FLAGS_teacher_id.empty()
+                                      ? FLAGS_a_top > 0 ? TOP_K : STUDENT
+                                      : TEACHER;
+        const auto actions_map = actions_type == STUDENT
+                                     ? ParseActionLabels(FLAGS_student_ac)
+                                     : actions_type == TOP_K
+                                         ? ParseActionLabels(FLAGS_top_ac)
+                                         : ParseActionLabels(FLAGS_teacher_ac);
+        const auto num_top_persons = actions_type == TOP_K ? FLAGS_a_top : -1;
+        const auto top_action_id = actions_type == TOP_K
+                                   ? std::distance(actions_map.begin(), find(actions_map.begin(), actions_map.end(), FLAGS_top_id))
+                                   : -1;
+        if (actions_type == TOP_K && (top_action_id < 0 || top_action_id >= static_cast<int>(actions_map.size()))) {
+            slog::err << "Cannot find target action: " << FLAGS_top_id << slog::endl;
+            return 1;
+        }
+
+        slog::info << "Reading video '" << video_path << "'" << slog::endl;
+        slog::info << "Loading Inference Engine" << slog::endl;
+        Core ie;
+        std::vector<std::string> devices = {FLAGS_d_act, FLAGS_d_fd, FLAGS_d_lm,
+                                            FLAGS_d_reid};
+        std::set<std::string> loadedDevices;
+        slog::info << "Device info: " << slog::endl;
+        for (const auto &device : devices) {
+            if (loadedDevices.find(device) != loadedDevices.end())
+                continue;
+            std::cout << printable(ie.GetVersions(device)) << std::endl;
+            loadedDevices.insert(device);
+        }
+
+        cv::Mat frame;
+        cv::VideoCapture cap;
+        video_path != "cam" ? cap.open(video_path) : cap.open(0);
+        if (!cap.read(frame)) {
+            GAPI_Assert(false && "Couldn't grab the very first frame");
+        }
+
+        /** NOTE: Empty strings. Need default constructor for cv::gapi::ie::Params */
+        cv::gapi::ie::Params<nets::FaceDetector> det_net({}, {}, {});
+        cv::gapi::ie::Params<nets::LandmarksDetector> landm_net({}, {}, {});
+        cv::gapi::ie::Params<nets::FaceReidentificator> reident_net({}, {}, {});
+        cv::gapi::ie::Params<nets::PersonDetActionRec> action_net({}, {}, {});
+
+        ActionDetectorConfig action_config;
+        ActionDetectionKernelInput ad_kernel_input;
+        if (!ad_model_path.empty()) {
```

Comment:
I'd instead introduce `operator+=` for a network package like @aDanPin did for graph arguments.

Reply:
Hmm. But it doesn't solve this problem.

Reply:
Actually it will, see on the right